github-api-call-inventory/github-api-call-inventory.ipynb

```diff
@@ -0,0 +1,765 @@
+    "    load_packit_yaml=5+1, \n",
+    "    builds_enabled=True, \n",
+    "    tests_enabled=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "a7159baf",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "6 chroots, config is called 'packit.yaml'\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "prs_per_hour_per_org = 11\n",
+       "total_is_project_private = 37\n",
+       "total_is_user_approved = 74\n",
+       "total_load_packit_yaml = 222\n",
+       "total_set_status = 84\n",
+       "grand_total = 417"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "print(\"6 chroots, config is called 'packit.yaml'\")\n",
+    "github_api_calls(\n",
+    "    chroots=6, \n",
+    "    rate_limit=5000, \n",
+    "    load_packit_yaml=5+1, \n",
+    "    builds_enabled=True, \n",
+    "    tests_enabled=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "a748a141",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "6 chroots, config is called '.packit.yaml', no tests\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "prs_per_hour_per_org = 26\n",
+       "total_is_project_private = 25\n",
+       "total_is_user_approved = 50\n",
+       "total_load_packit_yaml = 75\n",
+       "total_set_status = 36\n",
+       "grand_total = 186"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "print(\"6 chroots, config is called '.packit.yaml', no tests\")\n",
+    "github_api_calls(\n",
+    "    chroots=6, \n",
+    "    rate_limit=5000, \n",
+    "    load_packit_yaml=2+1, \n",
+    "    builds_enabled=True, \n",
+    "    tests_enabled=False\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "42af2a1a",
+   "metadata": {},
+   "source": [
+    "## Ways to improve things"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c05d101",
+   "metadata": {},
+   "source": [
+    "The main issue seems to be that `task.steve_jobs.process_message` re-creates the context of the pipeline by getting information from GitHub: checks if the project is private, if the user is approved to run the pipeline, and load the package config (`.packit.yaml`).\n",
+    "\n",
+    "All this could be avoided by looking up the pipeline when an event arrives (Copr events, testing farm events - when these are received there already needs to be a pipeline created, otherwise they are invalid), and getting this information from the pipeline.\n",
+    "\n",
+    "The only time it makes sense to load this from GitHub is when the pipeline is started, for example when the PR event is received."
```

Comment:
Is this observation correct? Would be nice if someone else could confirm this, b/c otherwise the numbers are way too wrong :sweat_smile:

Reply:
they shouldn't change within a pipeline run, right? people can become collaborators, be approved, packit.yaml can change, but NOT within a pipeline run: +1 to cache all of that data

this will also be a massive speedup